x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/action_form.tsx

```diff
@@ -360,6 +360,27 @@ export const ActionForm = ({
                 }}
                 onSelectConnector={(connectorId: string) => {
                   setActionIdByIndex(connectorId, index);
+                  const newConnector = connectors.find((connector) => connector.id === connectorId);
+                  if (newConnector && newConnector.actionTypeId) {
+                    const actionTypeRegistered = actionTypeRegistry.get(newConnector.actionTypeId);
+                    if (actionTypeRegistered.resetParamsOnConnectorChange) {
+                      const updatedActions = actions.map((_item: RuleAction, i: number) => {
```

Comment:
could we do a splice to just update the specific array item we care about? why do we need a map?

Reply:
`actions` is the state that contains all the actions added to a rule. The idea here is to update only the current action in case it has subtypes as they might have a different params model. As actions is a state, I'd like to avoid mutating it using splice and keep it as it is. 

Looking for best practices, this is the recommended way to go by the react team when replacing items in an array state https://react.dev/learn/updating-arrays-in-state#replacing-items-in-an-array